src/Avalonia.Controls/AutoCompleteBox.cs

```diff
@@ -2094,7 +2094,21 @@ private void RefreshView()
                 bool inResults = !(stringFiltering || objectFiltering);
                 if (!inResults)
                 {
-                    inResults = stringFiltering ? TextFilter(text, FormatValue(item)) : ItemFilter(text, item);
+                    if (stringFiltering)
+                    {
+                        inResults = TextFilter(text, FormatValue(item));
+                    }
+                    else
+                    {
+                        if (ItemFilter == null)
```

Comment:
sorry, just a tiny nit 
```suggestion
                        if (ItemFilter is null)
```

Reply:
@jmacato why do you consider this syntax as preferred?

Reply:
Well its the new syntax, and is the one were using here for new code as far as i recall

Reply:
There's a difference in the general case too: `is null` is a reference check, and `== null` may call an overload of `==` (which could return an unexpected result as well).